app/src/main/java/org/dhis2/uicomponents/map/geometry/mapper/featurecollection/MapDataElementToFeatureCollection.kt

```diff
@@ -0,0 +1,61 @@
+package org.dhis2.uicomponents.map.geometry.mapper.featurecollection
+
+import com.mapbox.geojson.Feature
+import com.mapbox.geojson.FeatureCollection
+import org.dhis2.data.dhislogic.CoordinateDataElementInfo
+import org.dhis2.uicomponents.map.geometry.bound.BoundsGeometry
+import org.dhis2.uicomponents.map.geometry.mapper.MapGeometryToFeature
+
+class MapDataElementToFeatureCollection(
+    private val mapGeometryToFeature: MapGeometryToFeature,
+    private val bounds: BoundsGeometry
+) {
+    private fun map(coordinateDataElementInfo: CoordinateDataElementInfo): Feature? {
+        bounds.initOrReset()
+
+        return mapGeometryToFeature.map(
+            coordinateDataElementInfo.geometry,
+            hashMapOf(
+                DE_NAME to coordinateDataElementInfo.dataElement.displayFormName()!!,
+                EVENT to coordinateDataElementInfo.event.uid()!!,
+                STAGE to coordinateDataElementInfo.stage.displayName()!!
+            ).apply {
+                coordinateDataElementInfo.enrollment?.let { enrollment ->
+                    put(TEI, enrollment.trackedEntityInstance()!!)
+                }
+            },
+            bounds
+        )
+    }
+
+    fun map(
```

Comment:
I think for you can make your testing easier if these two are separated classes like: MapRelationshipToFeature.kt and MapRelationshipToFeatureCollection.kt. I think you're missing some test cases in your unit test

Reply:
This will be also be changed in the other PR as the class name changed